src/web/js/beforePyret.js

```diff
@@ -1294,19 +1297,37 @@ $(function() {
       // this is blocks file. Open it with /blocks
       window.location.href = window.location.href.replace('editor', 'blocks');
     }
-    // NOTE(joe): Clearing history to address https://github.com/brownplt/pyret-lang/issues/386,
-    // in which undo can revert the program back to empty
-    CPO.editor.cm.setValue(c);
-    CPO.editor.cm.clearHistory();
+
+    if(!params["get"]["controlled"]) {
+      // NOTE(joe): Clearing history to address https://github.com/brownplt/pyret-lang/issues/386,
+      // in which undo can revert the program back to empty
+      CPO.editor.cm.setValue(c);
+      CPO.editor.cm.clearHistory();
+    }
+    else {
+      const hideWhenControlled = [
+        "#fullConnectButton",
+        "#logging",
+        "#logout"
+      ];
+      const removeWhenControlled = [
+        "#connectButtonli",
+      ];
+      hideWhenControlled.forEach(s => $(s).hide());
+      removeWhenControlled.forEach(s => $(s).remove());
+    }
+
   });
 
   programLoaded.fail(function() {
     CPO.documents.set("definitions://", CPO.editor.cm.getDoc());
   });
 
+  console.log("About to load Pyret: ", originalPageLoad, Date.now());
+
   var pyretLoad = document.createElement('script');
-  console.log(process.env.PYRET);
-  pyretLoad.src = process.env.PYRET;
+  console.log(window.PYRET);
+  pyretLoad.src = window.PYRET;
```

Comment:
Check if this is needed (was added for trying to embed as a webview directly in VScode, but that may not matter anymore because we are embedding a real built page)